services/blockchain-indexer/shared/dataService/business/pos/rewardsLocked.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const {
+	MySQL: { getTableInstance },
+	Exceptions: { InvalidParamsException },
+} = require('lisk-service-framework');
+
+const {
+	address: {
+		getLisk32AddressFromPublicKey,
+	},
+} = require('@liskhq/lisk-cryptography');
+
+const config = require('../../../../config');
+
+const { requestConnector } = require('../../../utils/request');
+
+const validatorsTableSchema = require('../../../database/schema/validators');
+
+const { getRewardTokenID } = require('../dynamicRewards/constants');
+
+const getValidatorsTable = () => getTableInstance(
+	validatorsTableSchema.tableName,
+	validatorsTableSchema,
+	config.endpoints.mysql,
+);
+
+const getRewardsLocked = async params => {
+	const response = {
+		data: [],
+		meta: {},
+	};
+
+	// Params must contain either address or name or publicKey
+	if (!Object.keys(params).some(param => ['address', 'name', 'publicKey'].includes(param))) {
+		throw new InvalidParamsException('One of the params (address, name or publicKey) is required.');
+	}
+
+	const tokenID = await getRewardTokenID();
+
+	// Process address
```

Comment:
Please leave a TODO here to simplify the implementation once the gateway issue related to `validParamPairings` is resolved.

Reply:
Applied.